static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,87 @@
+import {createContext, useCallback, useContext, useEffect, useState} from 'react';
+
+import {
+  DrawerBody,
+  DrawerContainer,
+  DrawerPanel,
+} from 'sentry/components/globalDrawer/components';
+import type {
+  DrawerConfig,
+  DrawerContext as DrawerUsageContext,
+  DrawerContextProps,
+} from 'sentry/components/globalDrawer/types';
+import {useLocation} from 'sentry/utils/useLocation';
+
+const DEFAULT_DRAWER_CONTEXT: DrawerContextProps = {
+  config: {
+    renderer: null,
+    options: {
+      closeOnEscapeKeypress: true,
+      closeOnOutsideClick: true,
+    },
+  },
+  openDrawer: () => {},
+  closeDrawer: () => {},
+};
+
+const DrawerContext = createContext<DrawerContextProps>(DEFAULT_DRAWER_CONTEXT);
+
+export function DrawerContextProvider({children}) {
```

Comment:
At this point, I probably wouldn't even call it a context provider (that's just an implementation detail), just call it `GlobalDrawer`